### PR TITLE
Issues/33 admin setting user institution in keycloack deployment

### PR DESCRIPTION
**Pull request by kostobog**
Only the last commit ([3c6ac92](https://github.com/kbss-cvut/record-manager/pull/34/commits/3c6ac92d5f4bb45164d4eb87df01cfea1b1959b2)) updating UserController is relevant. The other commits are from PR #31.

### CODE REVIEW

**Comment by kostobog** [comment]
@blcham Is this solution ok?

File: src/main/java/cz/cvut/kbss/study/rest/UserController.java [hunk 1]
```diff
@@ -101,6 +101,25 @@ public void removeUser(@PathVariable("username") String username) {
         }
     }
 
+    @ConditionalOnProperty(prefix = "security", name = "provider", havingValue = "oidc")
+    @PreAuthorize("hasRole('" + SecurityConstants.ROLE_ADMIN + "') or #username == authentication.name")
+    @PutMapping(value = "/{username}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void updateUserInstitution(@PathVariable("username") String username, @RequestBody User user,
```
Comment (blcham) [inline src/main/java/cz/cvut/kbss/study/rest/UserController.java:108]: @kostobog you are updating dedicated controller for "internal" authorization -- see `@ConditionalOnProperty` of this class !!! There is dedicated controller for "oidc" as stated here https://github.com/kbss-cvut/record-manager-ui/issues/12#issuecomment-1846132556 . It should be implemented there.

Comment (blcham) [reply src/main/java/cz/cvut/kbss/study/rest/UserController.java:108]: Otherwise it looks ok.

**Review by blcham — CHANGES_REQUESTED** [review]
See my comment.

File: src/main/java/cz/cvut/kbss/study/rest/UserController.java (hunk 1 above)
Comment (kostobog) [reply src/main/java/cz/cvut/kbss/study/rest/UserController.java:108]: Ok, I created a new PR #35 adding updateUserInstitution to OidcUserController.

**Comment by blcham** [comment]
@kostobog created a new PR https://github.com/kbss-cvut/record-manager/pull/35 adding updateUserInstitution to OidcUserController. Tests pass in PR https://github.com/kbss-cvut/record-manager/pull/35.

**Pull request closed without merging**